submissions/vovan-zt/TiniWorld2/index.js

```diff
@@ -0,0 +1,84 @@
+class WorldInhabitant {
+	constructor(species, name, gender, saying) {
+		this.species = species;
+		this.name = name;
+		this.gender = gender;
+		this.saying = saying;
+	}
+
+	getMeaning() {
+		return ["species", "name", "gender", "saying"]
+			.map((property) => {
+				if (property === "species") {
+					return this[property].toUpperCase();
+				}
+				return this[property];
+			})
+			.join("; ");
+	}
+}
+class Human extends WorldInhabitant {
+	constructor(name, gender, saying, friends) {
+		super("human", name, gender, saying);
+		this.legs = 2;
+		this.hands = 2;
+		this.friends = friends;
+	}
+
+	getMeaning() {
+		return `${super.getMeaning()};I have ${this.legs} legs and ${
+			this.hands
+		} hands; ${this.friends ? this.friends.name : "no friends"}`;
```

Comment:
This will work. Again you may wrap components (function/method call, string fragments, properties) into an array literal, and join it with e.g. `" "` (space).

Reply:
Do you propose to do this?

getMeaning() {
		return super.getMeaning() + [ 'I have', this.legs, 'legs and', this.hands, 'hands', ';', this.friends ? this.friends.name : "no friends"].join(' ');
	}

Reply:
`super.getMeaning()` can also perfectly be an array's element, for the sake of uniformity.

Reply:
But I liked my method better

Reply:
Can relate to this.
But code is written for other people.

Reply:
getMeaning() {
	return  [super.getMeaning(),
       `I have ${this.legs} legs`,
        `and ${this.hands} hands ;`,
         Array.isArray(this.friends) ? this.friends.join(', ') : "no friends"].join(' ');
	}

Reply:
Reads nice. If I had to add yet another property output, I would clearly see where is a good place to insert it among already existing.